Content.Server/Polymorph/Components/PolymorphableComponent.cs

```diff
@@ -0,0 +1,29 @@
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Polymorph;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+
+namespace Content.Server.Polymorph.Components
+{
+    [RegisterComponent]
+    public sealed class PolymorphableComponent : Component
+    {
+        /// <summary>
+        /// Whether or not the user is currently transformed
+        /// </summary>
+        [DataField("polymorphed")]
+        public bool Polymorphed = false;
+
+        /// <summary>
+        /// A list of all the polymorphs that the entity has.
+        /// Used to manage them and remove them if needed.
+        /// </summary>
+        [DataField("polymorphActions")]
+        public Dictionary<string, InstantAction> PolymorphActions = new();
```

Comment:
Not sure if this should be on a new component or not. At least make it nullable.